tests/database/test_journal_db.py

```diff
@@ -4,22 +4,38 @@
 
 
 @pytest.fixture
-def journal_db():
-    return JournalDB(MemoryDB())
+def memory_db():
+    return MemoryDB()
+
+
+@pytest.fixture
+def journal_db(memory_db):
+    return JournalDB(memory_db)
+
+
+def test_delete_removes_data_from_underlying_db_after_persist(journal_db, memory_db):
+    memory_db.set(b'1', b'test-a')
+
+    assert memory_db.exists(b'1') is True
+
+    journal_db.delete(b'1')
+    journal_db.persist()
```

Comment:
Maybe add a sanity check that the value is still in the memory db after the `delete` and before the `persist` call.

Reply:
Yes, sounds good.